searchcore/src/vespa/searchcore/proton/server/matchview.h

```diff
@@ -60,7 +62,7 @@ class MatchView {
     matching::MatchContext::UP createContext() const;
 
     std::unique_ptr<search::engine::SearchReply>
-    match(const ISearchHandler::SP &searchHandler,
+    match(const std::shared_ptr<ISearchHandler> &searchHandler,
```

Comment:
prefer passing shared pointers by value

Reply:
I normally do, but the usage pattern prevented the a simple change.
handler->match(handler, *searchRequest, *threadBundle)

Reply:
enable-shared-from-this is the better option. Will do later on.